src/test/kotlin/uk/gov/dluhc/printapi/mapper/aed/ReIssueAnonymousElectorDocumentMapperTest.kt

```diff
@@ -101,15 +107,29 @@ class ReIssueAnonymousElectorDocumentMapperTest {
         )
         given(aedMappingHelper.statusHistory(any())).willReturn(expectedStatusHistory)
 
+        val userId = aValidUserId()
+        val newElectoralRollNumber = aValidElectoralRollNumber()
+        val dto = buildReIssueAnonymousElectorDocumentDto(
+            sourceReference = sourceReference,
+            electoralRollNumber = newElectoralRollNumber,
+            deliveryAddressType = ERO_COLLECTION,
+            userId = userId
+        )
+
+        given(deliveryAddressTypeMapper.mapDtoToEntity(any())).willReturn(DeliveryAddressType.ERO_COLLECTION)
+
         val expected = previousAed.deepCopy().apply {
             certificateNumber = newCertificateNumber
             requestDateTime = FIXED_TIME
             issueDate = FIXED_DATE
             statusHistory = expectedStatusHistory.toMutableList()
+            electoralRollNumber = newElectoralRollNumber
+            delivery?.deliveryAddressType = DeliveryAddressType.ERO_COLLECTION
```

Comment:
How about checking that `delivery?.collectionReason` is also copied (which would be apt given the `DeliveryAddressType` here)?

Reply:
I had similar question, but then I checked the code and found out that the mapper was copying entire delivery from previousAed, so looks good.

Reply:
OK , will do .... 👍